kasa/smartdimmer.py

```diff
@@ -41,19 +41,37 @@ def brightness(self) -> int:
         return int(sys_info["brightness"])
 
     @requires_update
-    async def set_brightness(self, value: int):
-        """Set the new dimmer brightness level in percentage."""
+    async def set_brightness(self, brightness: int, transition: int = None):
+        """Set the new dimmer brightness level in percentage.
+
+        :param int transition: duration in milliseconds to transition to the new brightness (default: None)
+        """
         if not self.is_dimmable:
             raise SmartDeviceException("Device is not dimmable.")
 
-        if not isinstance(value, int):
-            raise ValueError("Brightness must be integer, " "not of %s.", type(value))
-        elif 0 <= value <= 100:
+        if not isinstance(brightness, int):
+            raise ValueError(
+                "Brightness must be integer, " "not of %s.", type(brightness)
+            )
+        elif not 0 <= brightness <= 100:
```

Comment:
Avoid `elif` construct here, as the error handling will raise an exception and it won't execute further anyway.